common/eth2/src/types.rs

```diff
@@ -1267,6 +1272,14 @@ pub struct ForkChoiceNode {
     pub execution_block_hash: Option<Hash256>,
 }
 
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(bound = "T: Serialize + serde::de::DeserializeOwned")]
+pub struct NextWithdrawalsResponse<T: Serialize + serde::de::DeserializeOwned> {
```

Comment:
Could use `ExecutionOptimisticFinalizedResponse` instead of adding this new type, but I don't mind too much either way.